src/App.vue

```diff
@@ -1,7 +1,10 @@
 <template>
   <main>
-    <c-header />
-    <c-table :messages="messages" />
+    <c-header @onUpdateMessages="updateMessages" @onUpdateRefreshTime="updateRefreshTime" />
+    <c-table :messages="messages" class="pt-32" />
+    <div class="flex w-full ">
+      <img class="mx-auto" src="@/assets/img/loading.gif" width="40rem" height="40rem" />
```

Comment:
Only show if it's sLoading=True. Also use a pure css spinner instead of a gif (the display is way better). Choose one of https://projects.lukehaas.me/css-loaders/ or search css spinner on the web. I would be nice is the spinner is above the table, you can use CSS position: absolute at put it in the middle of the page.